### DISCUSSION TITLE
Getting 504 Gateway Time-out error after 15 seconds on an api function (PRO plan)

### DISCUSSION THREAD

**Discussion by stajics** [vercel/community#3642] [Help] [answered]
### Problem Description

I am getting an 504 Gateway Time-out after 15 seconds for an api function endpoint . I have the PRO plan , what am I missing ? I tought the execution time is 60 sec https://vercel.com/docs/concepts/limits/overview ?

<img width="411" alt="Screenshot 2023-07-21 at 10 31 00 PM" src="https://github.com/vercel/vercel/assets/17711182/d08cba84-ef6f-439d-84b3-cb3b4ad3aa0f">

**Comment by dokeet** [comment]
I'm having the same issue with the Pro plan 😓

**Comment by emrecoban** [comment, marked as answer]
Did you configure the `vercel.json` file? https://vercel.com/docs/concepts/functions/serverless-functions#bundling-serverless-functions

`vercel.json` for Next.js (/app):
```json
{
  "functions": {
    "app/api/hello/route.js": {
      "memory": 3008,
      "maxDuration": 60
    },
    "app/api/another/route.js": {
      "memory": 1024,
      "maxDuration": 30
    }
  }
}
```
